### PR TITLE
Don't unschedule a resource if blocked by a cluster or syncer finalizer

### PR DESCRIPTION
**Pull request by jmprusi**
This PR changes the resource reconciler's behavior to avoid removing the sync state from a resource if a finalizer, either syncer or cluster, exists. 

fixes: https://github.com/kcp-dev/kcp/issues/1482

### CODE REVIEW

File: pkg/reconciler/workload/resource/resource_reconcile.go [hunk 1]
```diff
@@ -134,10 +135,25 @@ func computePlacement(ns *corev1.Namespace, obj metav1.Object) (annotationPatch
 	annotationPatch = map[string]interface{}{}
 	labelPatch = map[string]interface{}{}
 	for _, loc := range objLocations.Difference(nsLocations).List() {
-		// location was removed from namespace, but is still on the object
```
Comment (sttts) [inline pkg/reconciler/workload/resource/resource_reconcile.go:137]: leave comment?

**Comment by sttts** [comment]
@qiujian16 @jmprusi afaik we have no e2e test that removes something from a sync target. Can we add it?

**Comment by qiujian16** [comment]
I think we have that tests in scheduling test e2e, we evict a cluster which remove the service. I think we want to manually add a soft finalizer on that service and ensure it is not removed?

**Comment by jmprusi** [comment]
> I think we have that tests in scheduling test e2e, we evict a cluster which remove the service. I think we want to manually add a soft finalizer on that service and ensure it is not removed?

Yes, similar to the syncer e2e tests, where we add a soft finalizer and make sure that the resource is still there

**Pull request merged**